src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3033,8 +3047,88 @@ void initializeLoop()
             {
                 if (isObjectLoop)
                 {
-                    // PROTOTYPE(dataflow): Handle loops over object
-                    throw new NotImplementedException();
+                    // For i as Object = 3 To 6 step 2
+                    //    body
+                    // Next
+                    //
+                    // becomes ==>
+                    //
+                    // {
+                    //   Dim loopObj        ' mysterious object that holds the loop state
```

Comment:
>mysterious [](start = 46, length = 10)

Probably not the right word :)

Reply:
> Probably not the right word :)

Works for VB lowering, where I copied this comment from.

---
In reply to: [188118434](https://github.com/dotnet/roslyn/pull/26803#discussion_r188118434) [](ancestors = 188118434)